apis/nucleus/src/components/listbox/interactions/keyboard-navigation/keyboard-nav-rows.js

```diff
@@ -119,9 +119,8 @@ export default function getRowsKeyboardNavigation({
         }
         break;
       case KEYS.ESCAPE:
-        if (isModal()) {
-          cancel();
-        } else {
+        cancel();
```

Comment:
The old logic probably is not canceling selection in some scenarios (the else case), only in **modal** mode. So I'm guessing there exists some (special) scenarios/environments that selections cancelation might not happen on escape key press.

With new logic you are trying to **always** cancel selections first when escape key has been pressed -> which is maybe not the desired behaviour? 

but to assure if it's ok, probably Tobias or any other of our colleges who had worked on keyboard navigation logic may have more stronger/better opinion than me on this!